packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -159,7 +159,7 @@ export class SwapQuoter {
         this.orderbook = orderbook;
         this.expiryBufferMs = expiryBufferMs;
         this.permittedOrderFeeTypes = permittedOrderFeeTypes;
-        this._contractAddresses = getContractAddressesForChainOrThrow(chainId);
+        this._contractAddresses = options.contractAddresses || getContractAddressesForChainOrThrow(chainId);
```

Comment:
I wonder if it would be more useful to instead accept a `Partial<ContractAddresses>` and doing something like:
```ts
this._contractAddresses = {
    ...NULL_ADDRESSES,
    ...(() => {
        try {
            return getContractAddressesForChainOrThrow(chainId),
        } catch (err) {
            return {};
        }
    })(),
    ...options.contractAddresses
};
```
Though I guess the caller could just do that on their end. :man_shrugging: